### PR TITLE
Add spring data + validation

### PR DESCRIPTION
**Pull request by Iliailineya**

### CODE REVIEW

File: src/main/java/spring/exception/TaskNotFoundException.java [hunk 1]
```diff
@@ -0,0 +1,12 @@
+package spring.exception;
+
+import org.apache.log4j.Logger;
+
+public class TaskNotFoundException extends RuntimeException {
+    private static final Logger logger = Logger.getLogger(TaskNotFoundException.class);
```
Comment (YeaaaaH) [inline src/main/java/spring/exception/TaskNotFoundException.java]: no sens to use logger here

File: src/main/java/spring/exception/ProjectNotFoundException.java [hunk 2]
```diff
@@ -0,0 +1,12 @@
+package spring.exception;
+
+import org.apache.log4j.Logger;
+
+public class ProjectNotFoundException extends RuntimeException {
+    private static final Logger logger = Logger.getLogger(ProjectNotFoundException.class);
```
Comment (YeaaaaH) [inline src/main/java/spring/exception/ProjectNotFoundException.java]: no sens to use logger here

File: src/main/java/spring/exception/TaskReportNotFoundException.java [hunk 3]
```diff
@@ -0,0 +1,12 @@
+package spring.exception;
+
+import org.apache.log4j.Logger;
+
+public class TaskReportNotFoundException extends RuntimeException {
+    private static final Logger logger = Logger.getLogger(TaskReportNotFoundException.class);
```
Comment (YeaaaaH) [inline src/main/java/spring/exception/TaskReportNotFoundException.java]: no sens to use logger here

File: src/main/java/spring/model/dto/TaskDTO.java [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+@NoArgsConstructor
+public class TaskDTO {
+    @NotBlank
+    private String name;
+    @NotBlank
+    private String description;
+    @NotBlank
+    private String status;
+    @NotBlank
+    private String priority;
```
Comment (YeaaaaH) [inline src/main/java/spring/model/dto/TaskDTO.java]: use ENUM for priority and status

File: src/main/java/spring/model/dto/TaskReportDTO.java [hunk 5]
```diff
@@ -0,0 +1,18 @@
...
+import lombok.NoArgsConstructor;
+
+import java.time.LocalDate;
+
+@Data
+@AllArgsConstructor
+@NoArgsConstructor
+public class TaskReportDTO {
+    private Long taskId;
+    private LocalDate completionDate;
```
Comment (YeaaaaH) [inline src/main/java/spring/model/dto/TaskReportDTO.java:15]: do you need to track creational and update date?

File: src/main/java/spring/repository/ProjectRepository.java [hunk 6]
```diff
@@ -0,0 +1,10 @@
+package spring.repository;
+
+import org.springframework.data.jpa.repository.JpaRepository;
+import org.springframework.stereotype.Repository;
+import spring.model.Project;
+
+@Repository
```
Comment (YeaaaaH) [inline src/main/java/spring/repository/ProjectRepository.java]: redundant annotation

File: src/main/java/spring/repository/TaskReportRepository.java [hunk 7]
```diff
@@ -0,0 +1,10 @@
+package spring.repository;
+
+
+import org.springframework.data.jpa.repository.JpaRepository;
+import org.springframework.stereotype.Repository;
+import spring.model.TaskReport;
+
+@Repository
```
Comment (YeaaaaH) [inline src/main/java/spring/repository/TaskReportRepository.java]: redundant annotation

File: src/main/java/spring/repository/TaskRepository.java [hunk 8]
```diff
@@ -0,0 +1,9 @@
+package spring.repository;
+
+import org.springframework.data.jpa.repository.JpaRepository;
+import org.springframework.stereotype.Repository;
+import spring.model.Task;
+
+@Repository
```
Comment (YeaaaaH) [inline src/main/java/spring/repository/TaskRepository.java]: redundant annotation

File: src/main/java/spring/service/ProjectService.java [hunk 9]
```diff
@@ -0,0 +1,51 @@
...
+        Project project = getProjectById(id);
+        project.setName(projectDTO.getName());
+        project.setDescription(projectDTO.getDescription());
+        project.setStartDate(projectDTO.getStartDate());
+        project.setEndDate(projectDTO.getEndDate());
+        return repository.save(project).getId();
+    }
+
+    public void deleteProjectById(long id) {
+        if (repository.existsById(id)) {
```
Comment (YeaaaaH) [inline src/main/java/spring/service/ProjectService.java]: getProjectById?

File: src/main/java/spring/service/TaskReportService.java [hunk 10]
```diff
@@ -0,0 +1,51 @@
...
+
+    public Long updateTaskReport(long id, TaskReportDTO taskReportDTO) {
+        TaskReport taskReport = getTaskReportById(id);
+        taskReport.setDescription(taskReportDTO.getDescription());
+        taskReport.setCompletionDate(taskReportDTO.getCompletionDate());
+        return taskReportRepository.save(taskReport).getId();
+    }
+
+    public void deleteTaskReportById(long id) {
+        if (taskReportRepository.existsById(id)) {
```
Comment (YeaaaaH) [inline src/main/java/spring/service/TaskReportService.java]: getTaskReportById?

File: src/main/java/spring/service/TaskService.java [hunk 11]
```diff
@@ -0,0 +1,62 @@
...
+        task.setPriority(taskDTO.getPriority());
+        task.setCreatedDate(taskDTO.getCreatedDate());
+        task.setDueDate(taskDTO.getDueDate());
+        Project project = projectService.getProjectById(taskDTO.getProjectId());
+        task.setProject(project);
+        return taskRepository.save(task).getId();
+    }
+
+    public void deleteTaskById(long id) {
+        if (taskRepository.existsById(id)) {
```
Comment (YeaaaaH) [inline src/main/java/spring/service/TaskService.java]: getTaskReportById?

**Review by YeaaaaH — CHANGES_REQUESTED** [review]
can be improved

File: src/main/java/spring/model/dto/TaskReportDTO.java (hunk 5 above)
Comment (Iliailineya) [reply src/main/java/spring/model/dto/TaskReportDTO.java:15]: I think its redurant due to TaskDTO has this fields, am I wrong?

Comment (YeaaaaH) [reply src/main/java/spring/model/dto/TaskReportDTO.java:15]: if you have it, ofc, but i don't see them. Actually it's a question to you : do you need these fields or no. If you think it's okay not to track - just ignore

**Review by YeaaaaH — APPROVED** [review]
looks good

**Pull request merged**